src/coreclr/vm/jitinterface.cpp

```diff
@@ -8914,16 +8913,19 @@ CORINFO_CLASS_HANDLE CEEInfo::getDefaultEqualityComparerClassHelper(CORINFO_CLAS
         MODE_PREEMPTIVE;
     } CONTRACTL_END;
 
-    // Mirrors the logic in BCL's CompareHelpers.CreateDefaultEqualityComparer
-    // And in compile.cpp's SpecializeEqualityComparer
+    // Mirrors the logic in BCL's CompareHelpers.CreateDefaultEqualityComparer.
     TypeHandle elemTypeHnd(elemType);
 
-    // Mirrors the logic in BCL's CompareHelpers.CreateDefaultComparer
-    // And in compile.cpp's SpecializeComparer
-    //
-    // We need to find the appropriate instantiation
+    // We need to find the appropriate instantiation.
     Instantiation inst(&elemTypeHnd, 1);
 
+    // string
+    if (elemTypeHnd.IsString())
```

Comment:
This also needs a matching change in https://github.com/dotnet/runtime/blob/a7709e5b003c15c3f5555a04343f29b7b47f48cb/src/coreclr/tools/Common/TypeSystem/IL/Stubs/ComparerIntrinsics.cs#L44-L47

Reply:
Thanks. Wasn't sure if I'd tracked down all the places to be updated, hence the draft... was hoping CI would help highlight any misses. Will fix.